include/toml++/toml_print_to_stream.h

```diff
@@ -222,12 +222,31 @@ TOML_IMPL_NAMESPACE_START
 					return true;
 				};
 
+				static constexpr auto num_decimal_places = [](T number) {
+					// This prints numbers with up to 3 decimal places more coincely
+					constexpr double precision { 1e-4 };
+
+					T temp {};
+					int decimal_places {};
+					do {
+						number *= 10;
+						temp = number - int(number);
+
+						decimal_places++;
+					} while(temp > precision && decimal_places < std::numeric_limits<T>::digits10 + 1);
+
+					return decimal_places;
+				};
+
+				auto decimal_places = num_decimal_places(val);
+
+
 				#if TOML_FLOAT_CHARCONV
 				{
 					char buf[charconv_buffer_length<T>];
 					const auto res = hexfloat
-						? std::to_chars(buf, buf + sizeof(buf), val, std::chars_format::hex)
-						: std::to_chars(buf, buf + sizeof(buf), val);
+						? std::to_chars(buf, buf + sizeof(buf), val, std::chars_format::hex, decimal_places)
```

Comment:
You don't need to do this in the hexfloat case; full precision is assumed for hexfloats (that's sorta the whole point of them).